include/odp/api/spec/packet_flags.h

```diff
@@ -23,87 +23,114 @@ extern "C" {
 #include <odp/api/packet.h>
 
 /** @addtogroup odp_packet
- *  Boolean operations on a packet.
+ *  Operations on packet metadata flags.
  *  @{
  */
 
 /**
- * Check for packet errors
+ * Check for all errors in packet
  *
- * Checks all error flags at once.
+ * Check if packet parsing have found any errors in the packet. The level of
+ * error checking depends on parse configuration (e.g. included layers and
+ * checksums). Protocol layer functions (e.g. odp_packet_has_l3()) indicate
+ * which layers have been checked, and layer error functions
+ * (e.g. odp_packet_has_l3_error()) which layers have errors.
  *
- * @param pkt Packet handle
- * @retval non-zero packet has errors
- * @retval 0 packet has no errors
+ * @param pkt          Packet handle
+ *
+ * @retval non-zero    Packet has errors
+ * @retval 0           No errors were found
  */
 int odp_packet_has_error(odp_packet_t pkt);
 
 /**
- * Check for packet L2 errors
+ * Check for errors in layer 2
  *
- * check for all L2 errors
+ * Check if packet parsing have found any errors in layer 2 of the packet.
+ * Make sure that the layer is included into parse configuration.
  *
- * @param pkt Packet handle
- * @retval non-zero packet has L2 errors
- * @retval 0 packet has no L2 error
+ * @param pkt          Packet handle
+ *
+ * @retval non-zero    Packet has errors in layer 2
+ * @retval 0           No errors were found in layer 2
```

Comment:
Worth noting this may be because layer 2 parsing was not configured? Or are results undefined in this case?

Reply:
I didn't change this, since patch 4 gives the generic rule "... Other layer/protocol specific metadata have undefined values.". If application first says it does not need e.g. L4 parsing and then still asks for L4 metadata, those values are  undefined (implementation may leave it uninitialized).